src/System.Diagnostics.FileVersionInfo/src/System/Diagnostics/FileVersionInfo.Unix.cs

```diff
@@ -9,42 +14,195 @@ private FileVersionInfo(string fileName)
+                                // we can't differentiate these two cases, so given the rarity of explicitly setting an 
+                                // assembly's version number to 0.0.0.0, we assume that if it is 0.0.0.0 then the attribute 
+                                // wasn't specified and we use the file version.
+                                if (_productVersion == "0.0.0.0")
+                                {
+                                    _productVersion = _fileVersion;
+                                    _productMajor = _fileMajor;
+                                    _productMinor = _fileMinor;
+                                    _productBuild = _fileBuild;
+                                    _productPrivate = _filePrivate;
+                                }
+                            }
+                            break;
+                        case "System.Reflection.AssemblyProductAttribute":
+                            _productName = attrNameAndValue.Value;
+                            break;
+                        case "System.Reflection.AssemblyTrademarkAttribute":
+                            _legalTrademarks = attrNameAndValue.Value;
+                            break;
+                        case "System.Reflection.AssemblyTitleAttribute":
+                            _fileDescription = attrNameAndValue.Value;
+                            break;
+                    }
+                }
+            }
         }
 
-        // -----------------------------
-        // ---- PAL layer ends here ----
-        // -----------------------------
+        /// <summary>
+        /// Attempt to parse a CustomAttribute into its full name and the value of its single string parameter.
+        /// </summary>
+        /// <param name="reader">The metadata reader.</param>
+        /// <param name="attr">The attribute to parse.</param>
+        /// <param name="result">The name of the attribute and its parameter's value, if it could be parsed.</param>
+        /// <returns>true if the attribute could be parsed; otherwise, false.</returns>
+        private static bool TryGetAttributeNameAndStringArgument(
+            MetadataReader reader, CustomAttribute attr, out KeyValuePair<string, string> result)
+        {
+            var ctorHandle = attr.Constructor;
+            if (ctorHandle.Kind == HandleKind.MemberReference)
+            {
+                EntityHandle container = reader.GetMemberReference((MemberReferenceHandle)ctorHandle).Parent;
+                if (container.Kind == HandleKind.TypeReference)
+                {
+                    TypeReference attrType = reader.GetTypeReference((TypeReferenceHandle)container);
+                    string attrName = reader.GetString(attrType.Namespace) + "." + reader.GetString(attrType.Name);
+
+                    if (attr.Constructor.Kind == HandleKind.MemberReference)
```

Comment:
I believe this will fail for mscorlib.dll or System.Runtime.dll where the constructor handle is likely a MethodDefinition rather than a MemberReference.